### PR TITLE
REP-4596 Now with legacy slave manifest

### PR DESCRIPTION
**Pull request by adrianjgeorge**
Wait to do this one till after #58

### CODE REVIEW

File: myModules/repose_jenkins/manifests/modern_slave.pp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    chain => 'DOCKER-ISOLATION',
+    proto => 'all',
+    jump  => 'RETURN',
+  }
+
+  group { 'docker':
+    ensure => present,
+  }
+
+  $jenkins_home = '/var/lib/jenkins'
```
Comment (dmnjohns) [inline myModules/repose_jenkins/manifests/modern_slave.pp]: I don't love this being duplicated in all of the slave manifests. So what seems to be a common pattern is to create a params class and inherit that class wherever those params are needed. I know this is one variable, but I wonder if that is a pattern we would like to follow.

More information about inheritance can be found in the [Puppet inheritance docs](https://docs.puppet.com/puppet/4.10/lang_classes.html#inheritance).

File: myModules/repose_jenkins/manifests/legacy_slave.pp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  class { 'repose_jenkins::slave':
+    java_package => 'openjdk-7-jdk'
+  }
+
+  $jenkins_home = '/var/lib/jenkins'
+
+  class{"repose_maven":
+    user      => 'jenkins',
+    user_home => "${jenkins_home}",
+    require  => User["jenkins"],
```
Comment (adrianjgeorge) [inline myModules/repose_jenkins/manifests/legacy_slave.pp]: I'm guessing they never did or the copy pasta did something weird.

File: myModules/repose_jenkins/manifests/modern_slave.pp (hunk 1 above)
Comment (adrianjgeorge) [inline myModules/repose_jenkins/manifests/modern_slave.pp]: Yeah, this was what i was talking about in slack the other day when i said there was something still that i wasn't quite happy with.

Comment (adrianjgeorge) [inline myModules/repose_jenkins/manifests/modern_slave.pp]: I read that as discouragement of doing it via inheritance. I think I just want this: https://docs.puppet.com/puppet/4.10/lang_variables.html#qualified-variable-names

File: myModules/repose_jenkins/manifests/slave.pp [hunk 3]
```diff
@@ -1,21 +1,91 @@
...
+    gid        => 'jenkins',
+    groups     => $groups,
+    home       => $repose_jenkins::jenkins_home,
+    shell      => '/bin/bash',
+    managehome => true,
+  }
+
+
+  #all this key stuff may actually need to be moved back up
+  #dependening on wether or not master is the guy trying to push the packages to the repo
```
Comment (dmnjohns) [inline myModules/repose_jenkins/manifests/slave.pp]: We may be able to drop this comment? If master doesn't have any executors, then only the slaves should be running jobs which try to push to the package repo. So this seems legit.

Comment (adrianjgeorge) [inline myModules/repose_jenkins/manifests/slave.pp]: The problem is i'm not certain that master isn't the guy pushing the packages in the case of the maven build. it's possible that he copies them over locally then pushes them up. let me go look at that job again

Comment (adrianjgeorge) [inline myModules/repose_jenkins/manifests/slave.pp]: it looks like the slaves handle it, so we should be good

**Pull request merged**